core/arch/arm/kernel/mutex.c

```diff
@@ -60,6 +66,29 @@ static void __mutex_lock(struct mutex *m, const char *fname, int lineno)
 	}
 }
 
+static void __mutex_lock_recursive(struct recursive_mutex *m, const char *fname,
+				   int lineno)
+{
+	assert_have_no_spinlock();
+	assert(thread_get_id_may_fail() != -1);
+	assert(thread_is_in_normal_mode());
+
+	/* Thread ID has to fit in a short int */
+	COMPILE_TIME_ASSERT(CFG_NUM_THREADS <= SHRT_MAX);
+
+	if (atomic_load_short(&m->owner) == (short int)thread_get_id()) {
```

Comment:
looks like thread_id should be a short int, no?
or maybe `COMPILE_TIME_ASSERT(CFG_NUM_THREADS <= SHRT_MAX);` for consistency.

Reply:
You mean change `thread_get_id()` to return `short int` instead? Sounds reasonable to me. I can transform all `int` thread IDs to `short int` in a prior commit. @jenswi-linaro what do you think?

Reply:
Makes sense. Feel free to do that after this PR if that's more convenient.